### PR TITLE
chore: Clean up model package

### PR DESCRIPTION
**Pull request by Mrod1598**
### Proposed Change

- Add comments where there were todos
- Improve test coverage

##### Checklist
- [ ] Changes are tested
- [ ] CI has passed

### CODE REVIEW

File: model/mock_resource_store.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Code generated by mockery v2.31.1. DO NOT EDIT.
+// Code generated by mockery v2.28.1. DO NOT EDIT.
```
Comment (antonblock) [inline model/mock_resource_store.go:1]: Should `make install-tools` and regenerate with the same version

Comment (Mrod1598) [reply model/mock_resource_store.go]: Updated

File: model/resource_test.go [hunk 2]
```diff
@@ -15,12 +15,15 @@
 package model
 
 import (
+	"bytes"
 	"context"
 	"fmt"
 	"io/ioutil"
```
[the remark this answers is not in the document]
Comment (Mrod1598) [reply model/resource_test.go]: Updated

**Review by antonblock — APPROVED** [review]

**Review by BinaryFissionGames — COMMENTED** [review]
Awesome job on adding all these tests, just had a few questions on a couple TODOs. Trying to gauge if they should be removed or they are important.

File: model/configuration.go [hunk 3]
```diff
@@ -1138,14 +1134,6 @@ func (c *Configuration) IndexFields(index modelSearch.Indexer) {
 		destination.indexFields("destination", "destinationType", index)
 	}
 
-	// add pipeline fields
-	//
-	// TODO(andy): I was going to add pipeline:traces, pipeline:logs, and pipeline:metrics because I thought it would be a
-	// useful way to filter configurations. However, we need a ResourceStore implementation to call otelConfiguration and
-	// we don't have that here, even though indexing is actually done in the store. I think the best solution is to cache
-	// the output on the Spec and keep that up to date as any dependent sourceTypes and destinationTypes change. This will
-	// improve performance when comparing configurations and displaying the configuration in UI.
```
[the remark this answers is not in the document]
Comment (Mrod1598) [reply model/configuration.go:1147]: asked josh and he said just remove it. was from a year and a half ago.

File: model/parameter.go [hunk 4]
```diff
@@ -426,7 +424,6 @@ func (p ParameterDefinition) validateValidValues() error {
 		}
 	}
 
-	// TODO(dave+mitch): (validate mapToSubForm type)
```
[the remark this answers is not in the document]
Comment (Mrod1598) [reply model/parameter.go:429]: felt like it was related to the aws field that was added. think the validation was added.

**Review by BinaryFissionGames — APPROVED** [review]

**Pull request merged**